### PR TITLE
Convert to Create React App

### LINKED ISSUE #297: Convert to a Create React App app
[Create React App](https://github.com/facebook/create-react-app) takes a huge amount of hassle out of configuring and managing React projects. It is popular, widely adopted and very actively maintained. 

Unfortunately climate-explorer-frontend was created before CRA existed. It has the following problems:

- We have to manage webpack configuration ourselves. Webpack is a great thing, but configuring it is a substantial skill area unto itself, and it is often done poorly (including here).  This is a known problem, and one of the main reasons CRA was invented.
- Updates to React, JSX, ES6 (babel), and other infrastructure are painful or impossible (see `react-markdown` below for an example of why this is a problem).
- Our current webpack configuration blocks us adopting some packages developed for more mature versions of React. In particular, it seems to be impossible to get [`react-markdown`](https://github.com/rexxars/react-markdown) running in this webpack environment. `react-markdown` is the core of our external-text package.

Filing and acting on this issue is mainly motivated by the final point (`react-markdown`). I've been hitting pain points with our current infrastructure for a year or so, but never quite seriously enough to justify the effort to move over to CRA. Now I have. 

I have already transplanted our current app code into a CRA [project](https://github.com/pacificclimate/climate-explorer-frontend-cra). It works, and it now includes our external text package that uses `react-markdown`.

### PR DESCRIPTION
**Pull request by rod-glover**
Resolves #297 

To do:

- [x] update README: 
   - [x] env variables
   - [x] public folder: variable-options, external-text
   - [x] instructions for updating public folder contents
- [x] fix favicon
- [x] fix manifest
- [x] changes to Dockerfile: create and run a production build
- [x] fix variable-options.yaml loading
- [x] convert any remaining requires to imports
- [x] fix .eslintrc

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,113 +1,70 @@
...
-    "moduleNameMapper": {
-      "\\.(png|jpg|gif|ttf|eot|svg)$": "<rootDir>/test_support/jest/__mocks__/fileMock.js"
-    },
-    "unmockedModulePathPatterns": [
-      "fbjs",
-      "react",
-      "react-dom"
-    ]
-  },
+  "name": "climate-explorer-frontend-cra",
```
Comment (rod-glover) [inline package.json]: Wrong. Transfer over relevant descriptors from existing package.json

File: src/logo.svg [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 841.9 595.3">
```
Comment (rod-glover) [inline src/logo.svg]: Remove this thing. It's a React logo.

**Comment by rod-glover** [comment]
[Demo](http://docker-dev01.pcic.uvic.ca:30508/pcex/app/#/data/compare/ce_files). This runs a production build in docker, served with [serve](https://www.npmjs.com/package/serve), as suggested in the CRA documentation.

TODO:

- [ ] Set up dockerization so that contents of the build's `static` directory can be updated live.

**Comment by rod-glover** [comment]
[Demo](http://docker-dev01.pcic.uvic.ca:30508/pcex/app/#/data/compare/ce_files) has been updated.

@corviday , would you mind checking that the changes you introduced in the Time Series graph haven't been disturbed in this branch?

**Comment by corviday** [comment]
Time series graph is still working as intended! Do you want the rest of this reviewed?

**Comment by rod-glover** [comment]
@corviday , yes please.

File: .env [hunk 3]
```diff
@@ -0,0 +1,4 @@
+REACT_APP_TILECACHE_URL=https://tiles.pacificclimate.org/tilecache/tilecache.py
+REACT_APP_NCWMS_URL=https://services.pacificclimate.org/pcex/ncwms
+REACT_APP_CE_ENSEMBLE_NAME=ce
```
Comment (corviday) [inline .env]: Should probable be ce_files.

**Comment by corviday** [comment]
When I attempted to download and run the new version, I received the following error:

```
Failed to compile
./src/components/map-controllers/MapController.module.css (./node_modules/css-loader/dist/cjs.js??ref--6-oneOf-4-1!./node_modules/postcss-loader/src??postcss!./src/components/map-controllers/MapController.module.css)
Error: Cannot find module '@csstools/normalize.css'
This error occurred during the build time and cannot be dismissed.
```

Have I done it incorrectly?

**Comment by rod-glover** [comment]
Demo is running on docker-dev0, as above.

If you really want to install it, do `npm run reinstall`, which will delete ./node_modules and install it fresh. It is *not* compatible with the previous version, and must be reinstalled.

File: README.md [hunk 4]
```diff
@@ -9,42 +9,89 @@ Front end interface for the PCIC Climate Explorer. Node, React.js, Webpack, Babe
 
 Node.js >= 4.0
```
Comment (corviday) [inline README.md]: This needs an update; I was using version 8.10.0 and it was too old.

File: .env.development [hunk 5]
```diff
@@ -0,0 +1,2 @@
+REACT_APP_CE_BACKEND_URL=http://localhost:8000/api
+REACT_APP_CE_BASE_PATH=
```
Comment (corviday) [inline .env.development]: Does something need to be set here?

File: public/index.html [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+      Unlike "/favicon.ico" or "favicon.ico", "%PUBLIC_URL%/favicon.ico" will
+      work correctly both with client-side routing and a non-root public URL.
+      Learn how to configure a non-root public URL by running `npm run build`.
+    -->
+    <link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.5/css/bootstrap.min.css"
+          rel="stylesheet"
+          integrity="sha256-MfvZlkHCEqatNoGiOXveE8FIwMzZg4W85qfrfIFBfYc= sha512-dTfge/zgoMYpP7QbHy4gWMEGsbsdZeCXz7irItjcC3sPUFtf0kuFbDz/ixG7ArTxmDjLXDmezHubeNikyKGVyQ=="
+          crossorigin="anonymous">
+    <title>PCIC Climate Explorer (CRA)</title>
```
Comment (corviday) [inline public/index.html]: Does this need updated, similar to your comment above about updating the package name?

File: src/components/DataTool.js [hunk 7]
```diff
@@ -1,10 +1,32 @@
...
 
 import NavRoutes from './navigation/NavRoutes/NavRoutes';
 import SingleAppController from './app-controllers/SingleAppController/SingleAppController';
 import PrecipAppController from './app-controllers/PrecipAppController/PrecipAppController';
 import DualAppController from './app-controllers/DualAppController/DualAppController';
+import { loadVariableOptions } from '../core/util';
+import Await from './Await';
+
+
+function loadFakeSuccess() {
```
Comment (corviday) [inline src/components/DataTool.js]: What are these for? They only time they're used, they've been commented out.

File: src/components/map-controllers/SingleMapController/SingleMapController.js [hunk 8]
```diff
@@ -231,6 +231,7 @@ export default class SingleMapController extends React.Component {
         </Panel.Title>
         </Panel.Heading>
         <Panel.Body className={styles.mapcontroller}>
+          {/*Foo*/}
```
Comment (corviday) [inline src/components/map-controllers/SingleMapController/SingleMapController.js]: I'm sure there's some sort of style black magic to having this here, but a comment explaining what would be nice. :)

File: src/core/__tests__/util-test.js [hunk 9]
```diff
@@ -189,9 +190,15 @@ var mockAPI = require('../__test_data__/sample-API-results');
...
-    xit('returns the requested option', function () {});
+    it('returns undefined for nonexistent variables', function () {
+      expect(util.getVariableOptions('foo', 'bar')).toBeUndefined();
+    });
+    it('returns undefined for nonexistent options', function () {
+      expect(util.getVariableOptions('tasmin', 'fuggle')).toBeUndefined();
+    });
+    it('returns the requested option', function () {
+      expect(util.getVariableOptions('tasmin', 'decimalPrecision')).toBe(1);
+    });
```
Comment (corviday) [inline src/core/__tests__/util-test.js]: Nice! It's great to be able to test these now!

File: .env (hunk 3 above)
Comment (corviday) [reply .env]: There isn't a `ce` ensemble and hasn't been for a while; the current ensemble is called `ce_files`.

But we set the ensemble via portal routing, so it doesn't actually _matter_ that this environment variable points to a nonexistent ensemble, I think. Possibly this environment variable is overdue to be removed entirely, but I'm not sure if that would have side effects.

File: .env.development (hunk 5 above)
Comment (rod-glover) [reply .env.development]: No. This sets it to the empty string, which is what's required in a typical dev run.

File: src/components/map-controllers/SingleMapController/SingleMapController.js (hunk 8 above)
Comment (rod-glover) [reply src/components/map-controllers/SingleMapController/SingleMapController.js]: Absolutely not. Cruft of the worst and most hackneyed order. Gone!

File: .env (hunk 3 above)
Comment (rod-glover) [reply .env]: Right. I've put in an [issue](https://github.com/pacificclimate/climate-explorer-frontend/issues/301).

**Comment by rod-glover** [comment]
The last few minor changes are good, app still works.

**Pull request merged**